src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -537,11 +533,31 @@ internal sealed override Cci.CallingConvention CallingConvention
 
         #region Syntax
 
-        internal SyntaxNode BodySyntax
+        internal (BlockSyntax, ArrowExpressionClauseSyntax) Bodies
         {
             get
             {
-                return (this.bodySyntaxReferenceOpt == null) ? null : this.bodySyntaxReferenceOpt.GetSyntax();
+                switch (SyntaxNode)
+                {
+                    case BaseMethodDeclarationSyntax method:
+                        return (method.Body, method.ExpressionBody);
+
+                    case AccessorDeclarationSyntax accessor:
+                        return (accessor.Body, accessor.ExpressionBody);
+
+                    case ArrowExpressionClauseSyntax arrowExpression:
+                        Debug.Assert(arrowExpression.Parent.Kind() == SyntaxKind.PropertyDeclaration ||
+                                     arrowExpression.Parent.Kind() == SyntaxKind.IndexerDeclaration ||
+                                     this is SynthesizedClosureMethod);
+                        return (null, arrowExpression);
+
+                    case BlockSyntax block:
+                        Debug.Assert(this is SynthesizedClosureMethod);
+                        return (block, null);
+
+                    default:
+                        return (null, null);
```

Comment:
This default makes me uneasy. Can it be reached? Could the switch be made complete (throw in default case)?

Reply:
> This default makes me uneasy. Can it be reached? Could the switch be made complete (throw in default case)?

Certainly synthesized methods don't have body in syntax. One might say they shouldn't inherit from this class, but they do. So, there is nothing wrong in getting here.

---
In reply to: [170045622](https://github.com/dotnet/roslyn/pull/24918#discussion_r170045622) [](ancestors = 170045622)